src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -29,6 +29,7 @@
 
   private final static Type MESSAGE_TYPE = new TypeToken<Map<String, String>>(){}.getType();
   private Gson gson;
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
why move this out of the init?

I think, either we keep it in the init and set it like it was before, or, if it's going to live up here, making it private and final.

However I don't know if you are supposed to use datastore like that, so I think perhaps we can just keep it in the init  method as it was before.

Reply:
@alekseychaikovsky Oh I don't think I ever put it in the init but I should have, fixed it!